### PR TITLE
(#20) added docs for region

### PR DESCRIPTION
**Pull request by cliffle**

### CODE REVIEW

File: content/writing_tests/screenshot.md [hunk 1]
```diff
@@ -64,15 +64,122 @@ To reproduce this scenario, you need to capture screenshots of the egg and the p
...
         testCase.saveResult();
     }
-})().then(done);
+})();
+{{< /highlight >}}
+
+## How to Control the Mouse in a Sakuli Test
+
+Before using a `region` it is best to instantiate a `region` over the entire display, to use it multiple times shown in
+the examples above.
```
Comment (s1hofmann) [inline content/writing_tests/screenshot.md]: This sentence somehow feels missplaced.

The heading says _"How to Control the Mouse..."_ but does not mention it at all in the first sentence.
Maybe something like

"Sakuli provides two ways to steer mouse movement. Either via coordinates, or via screenshot data. In order to search for image data on your screen you have to instantiate a `Region` spanning the entire display to access it's image content."

will outline the big picture of mouse movement while explaining why a `Region` is required.

File: content/writing_tests/screenshot.md [hunk 2]
```diff
@@ -64,15 +64,122 @@ To reproduce this scenario, you need to capture screenshots of the egg and the p
...
+
+### Moving the mouse to a screenshot
+
+To find a screenshot on the screen, you can use `find` or `waitForImage`. With `waitForImage` you can specify a timeout
+in which Sakuli continuously searches for the screenshot. This is useful, when e.g. starting a program like excel.
+
+{{< highlight typescript >}}
+await screen.find("search_this_screenshot.jpg")
```
Comment (s1hofmann) [inline content/writing_tests/screenshot.md:100]: We should add a line

```js
const screen = new Region();
```

to all samples for consistency.

File: content/writing_tests/screenshot.md [hunk 3]
```diff
@@ -64,15 +64,122 @@ To reproduce this scenario, you need to capture screenshots of the egg and the p
...
+    .mouseMove();
+    
+//searches the screen for 5 seconds for the excel homescreen
+await screen.waitForImage("excel_homescreen.jpg", 5000)
+    .mouseMove();
+{{< /highlight >}} 
+
+`find` and `waitForImage` returns the found screenshot as a region, so chaining as follows might not work, if the second
```
Comment (s1hofmann) [inline content/writing_tests/screenshot.md]:
```suggestion
`find` and `waitForImage` will return a `Region` for matched screenshots, so chaining as follows might not work, if the second
```

File: content/writing_tests/screenshot.md [hunk 4]
```diff
@@ -64,15 +64,122 @@ To reproduce this scenario, you need to capture screenshots of the egg and the p
...
+screenshot is not in the region of the first one.
```
Comment (s1hofmann) [inline content/writing_tests/screenshot.md]:
```suggestion
screenshot is not within the `Region` of the first one.

File: content/writing_tests/screenshot.md [hunk 5]
```diff
@@ -64,15 +64,122 @@ To reproduce this scenario, you need to capture screenshots of the egg and the p
...
+{{< highlight typescript >}}
+await screen.find("google_search.png")
+    .mouseMove()
+    .find("browser_address_bar.jpg")
+    .mouseMove();
+{{< /highlight >}}
+
+### Moving the mouse relative to current position
+A method to change the position is to use the `move()` method. This allows us to shift the used region, e.g.
```
Comment (s1hofmann) [inline content/writing_tests/screenshot.md]:
```suggestion
One way to change a `Region`'s position is to use the `move(x, y)` method. This allows us to shift a region in `x` and / or `y` direction, e.g.
```

File: content/writing_tests/screenshot.md [hunk 6]
```diff
@@ -64,15 +64,122 @@ To reproduce this scenario, you need to capture screenshots of the egg and the p
...
+    .move(50,50)
+    .mouseMove()
+    .click();
+})();
+{{< /highlight >}}
+
+
+### Moving the Mouse to an Absolute Position
+
+As mentioned before, you need to create a region with a width and heigth of 1 and your x/y coordinates.
```
Comment (s1hofmann) [inline content/writing_tests/screenshot.md]: Where did we mention to create a region with a width and heigth of 1 and our x/y coordinates? 😃

**Review by s1hofmann — CHANGES_REQUESTED** [review]

**Comment by s1hofmann** [comment]
I like the docs, but there are some methods missing:

- `left`
- `right`
- `above`
- `below`
- `grow`

File: content/writing_tests/screenshot.md (hunk 6 above)
Comment (cliffle) [reply content/writing_tests/screenshot.md]: I think i removed the part where it was mentioned and forgot to update this :see_no_evil:

File: content/writing_tests/screenshot.md [hunk 7]
```diff
@@ -64,15 +64,150 @@ To reproduce this scenario, you need to capture screenshots of the egg and the p
...
+const screen = new Region();
+await new Region(500, 500, 100, 100).left(200)
+//this returns a new region with (300/500/200/100)
+{{< /highlight >}}
+
+You can also manipulate the current `Region` with `setX(number)`, `setY(number)`, `setH(number)`, `setW(number)` or
+`grow(range)`. With `grow` you can expand the current region with _range_ px in all four directions.
+{{< highlight typescript >}}
+await screen = new Region();
+await new Region(500, 500, 100, 100).grow(100);
```
Comment (s1hofmann) [inline content/writing_tests/screenshot.md:132]: Number are a bit off here :)

**Review by s1hofmann — CHANGES_REQUESTED** [review]

**Review by s1hofmann — APPROVED** [review]

**Pull request merged**